pkg/operator/ceph/csi/spec.go

```diff
@@ -272,7 +293,11 @@ func StartCSIDrivers(namespace string, clientset kubernetes.Interface, ver *vers
 		logger.Warning("CSI Block volume expansion requires Kubernetes version >=1.16.0")
 	}
 
-	logLevel := os.Getenv("CSI_LOG_LEVEL")
+	logLevel, err := k8sutil.GetOperatorSetting(clientset, controllerutil.OperatorSettingConfigMapName, "CSI_LOG_LEVEL", "")
```

Comment:
When we have a lot of settings accessing the same configmap, I wonder if we should have a single call to load the configmap, then a helper that takes a configmap that is already loaded. Then we would lessen how many times the configmap is loaded. Let's keep that in mind for a future PR, but I don't think we need to worry about it for now.

Reply:
I believe this will be taken care automatically once we move on to controller-runtime which handles these situations.

Reply:
> @travisn @leseb Using ConfigMap to set operator cmd flags looks tricky.
> 
> Fetching the ConfigMap requires Kubernetes Client with correct context. This Context object is only created when the operator starts. Looks like a loop to me.
> 
> What do you suggest? (we also need to make it generic enough for everyone else to consume)

The command line flags are not really useful IMO. We really only need to support two ways of specifying the settings: env vars and the configmap. If we don't find the setting in the configmap, we look in the env vars. Is there a specific setting where you see the command line param causing issues? If so, we can probably remove the command line processing for that param.

Reply:
Removed unnecessary flags and configured CSI Params using ConfigMap or Env Var whichever is available.